src/data/datasets/representation/validation.py

```diff
@@ -0,0 +1,288 @@
+from cmath import pi
+from pathlib import Path
+from typing import Tuple, Union, Optional
+
+import torch
+import pandas as pd
+from albumentations import BasicTransform
+from albumentations.core.composition import BaseCompose
+
+from src.data.datasets.base import ImageDataset
+
+
+class RetrievalDataset(ImageDataset):
+    """Dataset for query->relevant benchmarking
+    Dataset for image retrieval where gallery consists of query, relevant and gallery items(optional).
+    The searches are made by queries while looking for relevant items in the whole set of items.
+    Where gallery items are treated non-relevant.
```

Comment:
Looks cluttered and also the first line repeats dataset definition. Better to leave just one line `Dataset for image retrieval validation ...`